packages/hierarchy-builder/src/test/internal/operators/grouping/BaseClassGrouping.test.ts

```diff
@@ -0,0 +1,298 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { expect } from "chai";
+import sinon from "sinon";
+import { HierarchyNode } from "../../../../hierarchy-builder/HierarchyNode";
+import * as baseClassGrouping from "../../../../hierarchy-builder/internal/operators/grouping/BaseClassGrouping";
+import { ECClass, IMetadataProvider } from "../../../../hierarchy-builder/Metadata";
+import { createGetClassStub, createTestNode, TStubClassFunc } from "../../../Utils";
+
+describe("BaseClassGrouping", () => {
+  const metadataProvider = {} as unknown as IMetadataProvider;
+  const ecClass = {
+    fullName: "TestSchema.TestParentClass",
+    label: "TestParentClass",
+  } as unknown as ECClass;
+  let stubClass: TStubClassFunc;
+  beforeEach(() => {
+    stubClass = createGetClassStub(metadataProvider).stubClass;
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe("getBaseClassGroupingECClasses", () => {
+    it("extracts ECClasses in the order: the most base -> lesser base -> least base ECClass", async () => {
+      const nodes: HierarchyNode[] = [
+        createTestNode({
+          key: { type: "instances", instanceKeys: [{ className: "TestSchema.TestClass", id: "0x1" }] },
+          params: {
+            grouping: {
+              byBaseClasses: {
+                fullClassNames: ["TestSchema.TestParentClass2", "TestSchema.TestParentClass1", "TestSchema.TestParentClass3"],
+              },
+            },
+          },
+        }),
+      ];
+      stubClass({
+        schemaName: "TestSchema",
+        className: "TestParentClass1",
+        isEntityClass: () => true,
+        isRelationshipClass: () => true,
+      });
+      stubClass({
+        schemaName: "TestSchema",
+        className: "TestParentClass2",
+        isEntityClass: () => true,
+        isRelationshipClass: () => true,
+        is: async () => true,
+      });
+      stubClass({
+        schemaName: "TestSchema",
+        className: "TestParentClass3",
+        isEntityClass: () => true,
+        isRelationshipClass: () => true,
+        is: async (className) => className === "TestSchema.TestParentClass1",
+      });
+
+      const result = await baseClassGrouping.getBaseClassGroupingECClasses(metadataProvider, nodes);
+      expect(result[0].fullName).to.eq("TestSchema.TestParentClass1");
+      expect(result[1].fullName).to.eq("TestSchema.TestParentClass3");
+      expect(result[2].fullName).to.eq("TestSchema.TestParentClass2");
```

Comment:
Could we:
- simplify class names to "Class1", "Class2", "Class3"
- make "1" the most base class and "3" the most derived one

Reply:
Changed to your suggestion.